react/core/main.tsx

```diff
@@ -137,7 +156,15 @@ const render = (
     ? ((disableSSR || created
         ? renderDOM<HTMLDivElement>(root, elem)
         : hydrate(root, elem)) as Element)
-    : renderToStringWithData(root).then(({ markup, renderTimeMetric }) => ({
+    : !disableSSQ
+    ? renderToStringWithData(root).then(({ markup, renderTimeMetric }) => ({
```

Comment:
Nesting

Actually, take a look here, I stole this from a Bruno Abreu WIP, and he reorganized it into a much better way:

```
if (!canUseDOM) {
    return renderToStringWithData(root).then(
      ({ markup, renderTimeMetric }) => ({
        markups: getMarkups(name, markup),
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        maxAge: cacheControl!.maxAge,
        page,
        renderTimeMetric,
      })
    )
  }

  if (disableSSR || created) {
    renderDOM<HTMLDivElement>(root, elem)
  } else {
    hydrate(root, elem)
  }
```

The early return for SSR makes the code simpler to follow. Will also avoid a nightmare rebase for my branch 😂